src/anemoi/models/models/encoder_processor_decoder.py

```diff
@@ -32,24 +31,30 @@ class AnemoiModelEncProcDec(nn.Module):
     def __init__(
         self,
         *,
-        config: DotConfig,
+        config: DotDict,
         data_indices: dict,
-        graph_data: HeteroData,
+        graph_data: dict,
     ) -> None:
         """Initializes the graph neural network.
 
         Parameters
         ----------
         config : DictConfig
             Job configuration
-        graph_data : HeteroData
+        graph_data : dict
             Graph definition
         """
         super().__init__()
 
         self._graph_data = graph_data
-        self._graph_name_data = config.graph.data
-        self._graph_name_hidden = config.graph.hidden
+        self._graph_name_hidden = config.graphs.hidden_mesh.name
+        self._graph_mesh_names = [name for name in graph_data if isinstance(name, str)]
+        self._graph_input_meshes = [
```

Comment:
These need a comment to explain what is being done here.

Probably should be a method too that takes `graph_data` as an input and processes it.

Reply:
From reading the code all the way to the end, it seems that these are filtering the `names` / `keys` of the graphs. This should be reflected in the variable names.